osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -187,7 +229,25 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
     r["odrops"] = INTEGER(0);
     r["collisions"] = BIGINT_FROM_UINT32(ifd->ifi_collisions);
     r["last_change"] = BIGINT_FROM_UINT32(ifd->ifi_lastchange.tv_sec);
-#endif
+    if (context.isColumnUsed("link_speed")) {
+      int fd = socket(AF_INET, SOCK_DGRAM, 0);
+      if (fd >= 0) {
+        struct ifmediareq ifmr = {};
+        memcpy(ifmr.ifm_name, addr->ifa_name, sizeof(ifmr.ifm_name));
+        ifmr.ifm_ulist = new int[ifmr.ifm_count];
+        if (ioctl(fd, SIOCGIFMEDIA, &ifmr) >= 0) {
+          if (IFM_TYPE(ifmr.ifm_active) == IFM_ETHER) {
+            int ifmls = get_linkspeed(IFM_SUBTYPE(ifmr.ifm_active));
+            if (ifmls > 0) {
+              r["link_speed"] = BIGINT_FROM_UINT32(ifmls);
+            }
+          }
+        }
+      }
+      close(fd);
```

Comment:
Technically you this only has to be called if `fd >= 0`, but I don't think it'll hurt anything to call it always.